openedx/core/djangoapps/credit/models.py

```diff
@@ -176,6 +180,22 @@ class CreditRequirementStatus(TimeStampedModel):
     # the grade to users later and to send the information to credit providers.
     reason = JSONField(default={})
 
+    @classmethod
+    def get_status(cls, requirement, username):
+        """ Get credit requirement statuses of given requirement and username
+
+        Args:
+            requirement(CreditRequirement): The identifier for a requirement
+            username(str): username of the user
+
+        Returns:
+            CreditRequirementStatus object
+        """
+        try:
+            return cls.objects.get(requirement=requirement, username=username)
```

Comment:
if one requirement has one status for a user then there should be unique_together('username', 'requirement') clause in Meta class of the CreditRequirementStatus model.